public/javascripts/SVLabel/src/SVLabel/keyboard/Keyboard.js

```diff
@@ -7,6 +7,19 @@
  */
 function Keyboard (svl, canvas, contextMenu, googleMap, ribbon, zoomControl) {
     var self = this;
+
+    /**
+     * fix for the shift-getting-stuck bug.
+     * this is a documented issue, see here:
+     * https://stackoverflow.com/questions/11225694/why-are-onkeyup-events-not-firing-in-javascript-game
+     *
+     * essentially what's going on is that JS sometimes fires a final keydown after a keyup.
+     * (usually happens when multiple events are fired)
+     * so the log would look like keydown:shift, keydown: shift, keyup: shift, keydown: shift.
+     * To fix this, we note the last time that shift was let go, then
+     * ignore any keydown events that were made BEFORE shift was let go, but are executing AFTER.
```

Comment:
Is this description of the fix still up to date?

Reply:
added a blurb about the 100ms stuff. Yeah, the old problem still exists, and this info is still relevant.